math/mathcore/src/TRandom3.cxx

```diff
@@ -132,21 +133,24 @@ Double_t TRandom3::Rndm()
    y ^= ((y << 15) & kTemperingMaskC );
    y ^=  (y >> 18);
 
-   // 2.3283064365386963e-10 == 1./(max<UINt_t>+1)  -> then returned value cannot be = 1.0
-   if (y) return ( (Double_t) y * 2.3283064365386963e-10); // * Power(2,-32)
-   return Rndm();
+   // 2.3283064365386963e-10 == 1. / (max<UInt_t> + 1) -> then returned value cannot be = 1.0
+   return static_cast<Double_t>(y * 2.3283064365386963e-10);
```

Comment:
We can add the +1 here after converting to double to avoid overflows in the uint:
```
return (static_cast<Double_t>(y) + 1)* 2.3283064365386963e-10)
```

Reply:
The overhead of adding a +1 should be minimal. We could test running `tutorial/math/testRandom.C`

Reply:
Yes, I'm testing the implementation with +1 ATM.

Reply:
This is the result:
```
                   TRandom3    TRandom3 with +1
Rndm..............   20.000      15.000
RndmArray.........   10.000      10.000
Gaus..............   60.000      55.000
Rannor............   40.000      40.000
Landau............   40.000      40.000
Exponential.......   35.000      35.000
Binomial(5,0.5)...  125.000     130.000
Binomial(15,0.5)..  370.000     385.000
Poisson(3)........   90.000      95.000
Poisson(10).......  205.000     215.000
Poisson(70).......  295.000     300.000
Poisson(100)......  295.000     300.000
GausTF1...........  180.000     180.000
LandauTF1.........  175.000     175.000
```